.NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishTimePeriodParserConfiguration.cs

```diff
@@ -121,11 +121,22 @@ public bool GetMatchedTimeRange(string text, out string timex, out int beginHour
                 return false;
             }
 
-            var parseResult = TimexUtility.ResolveTimeOfDay(timeOfDay);
-            timex = parseResult.Timex;
-            beginHour = parseResult.BeginHour;
-            endHour = parseResult.EndHour;
-            endMin = parseResult.EndMin;
+            if ((Options & DateTimeOptions.TasksMode) != 0)
+            {
+                var parseResultTaskMode = TimexUtility.ResolveTimeOfDayTaskMode(timeOfDay);

```

Comment:
Shouldn't just the specific resolve call inside the if/else?

Reply:
changes made and commited